apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -177,89 +177,107 @@ resolve_loop(Begin, Workers, Fun, FailMsg, Zone) ->
 -spec(start_worker(protocol(), upstream(), binary()) -> pid()).
 start_worker(Protocol, Upstream, Request) ->
     Pid = self(),
+    Begin = erlang:monotonic_time(),
     proc_lib:spawn(fun () ->
-        init_worker(Protocol, Pid, Upstream, Request)
+        Address = upstream_to_binary(Upstream),
+        init_worker(Protocol, Pid, {Upstream, Address}, Request),
```

Comment:
We ever can move all metrics and logging here and make all other functions much cleaner, right?